File: Week 3/C++ Codes/DeleteWithoutHeadPointer.cpp
#include<bits/stdc++.h>
using namespace std;

struct Node {
  int data;
  struct Node *next;
  Node(int x) {
    data = x;
    next = NULL;
  }
}*head;

// Swap the values
void deleteNode(Node *node)
{
    // Your code here
    if (node == NULL or node->next == NULL)
        return;
    int val = node->data;
    node->data = node->next->data;
    node->next->data = val;
    node->next = node->next->next;
}

// Copy the values
void deleteNode(Node *node)
{
   // Your code here
   node->data = node->next->data;
   node->next = node->next->next;
}

File: Week 3/Python Codes/DeleteWithoutHeadPointer.py
# Node Class
class Node:
    def __init__(self, data):   # data -> value stored in node
        self.data = data
        self.next = None

# swap the values
def deleteNode(curr_node):
    if not curr_node or not curr_node.next:
        return
    curr_node.data, curr_node.next.data = curr_node.next.data, curr_node.data
    curr_node.next = curr_node.next.next

# copy the values
def deleteNode(curr_node):
    if not curr_node or not curr_node.next:
        return
    curr_node.data = curr_node.next.data
    curr_node.next = curr_node.next.next
    